test/mainline_test.go

```diff
@@ -339,6 +349,21 @@ func removeInstallResourceCR(c client.Client, name string, ctx context.Context)
 	Expect(err).NotTo(HaveOccurred())
 	err = c.Delete(ctx, instance)
 	Expect(err).NotTo(HaveOccurred())
+	// Need to wait here for Installation resource to be fully deleted prior to cancelling the context
+	// which will in turn terminate the operator. Race conditions can occur otherwise that will leave the
+	// Installation resource intact while the operator is no longer running which will result in test failures
+	// that try to create an Installation resource of their own
+	Eventually(func() error {
+		installation := &operator.Installation{
+			TypeMeta:   metav1.TypeMeta{Kind: "Installation", APIVersion: "operator.tigera.io/v1"},
+			ObjectMeta: metav1.ObjectMeta{Name: name},
+		}
+		err = GetResource(c, installation)
+		if kerror.IsNotFound(err) || kerror.IsGone(err) {
+			return nil
+		}
+		return fmt.Errorf("\"%s\" installation still exists", name)
```

Comment:
I think we should return the err if it's non-nil but also not a 404

Reply:
Done